rfcs/as-prop-and-custom-component.md

```diff
@@ -0,0 +1,154 @@
+
+```jsx
+<Button type="submit" formtarget="_blank">Submit</Button>
+```
+
+For some components, there is not a DOM element strongly associated with the component. In those case, the default value is "div", but in the cases where the as property change often, on the Box
+component for instance, we simply provide all the possible HTML attributes for that component.
+
+```jsx
+<Box id="myBox">
+```
+
+Finally, if you try to provide a Custom Component to the as property of a Box component, the property of that Custom Component won't be proposed by default through the intellisense and won't be type checked. 
+
+```jsx
+<TextLink as={RouterLink} to="/flight-details"> // The property isn't provided via intellisense, but can still be added
+    Flight details
+</TextLink>
+```
+
+It's ok in some cases, but if your developing a component that is going to be reused a lot, you should look at Option 2. 
+
+### Option 2: Using forwardRef from @orbit-ui/react-components
+
+`forwardRef` is the recommended approach as it ensures your components forwards their reference properly. Orbit defined its own `forwardRef` that adds the HTML or React properties to the component.
+
+>  Note ⚠️: You need to use forwardRef from @orbit-ui not react
+
+Example of the way the link is exported by default : 
+```jsx
+import { forwardRef } from "@orbit-ui/react-components";
+
+// This way of exporting is not specifying a type to use as its default as property. So we add all the HTML properties
+export const Link = forwardRef<InnerLinkProps>((props, ref) => (
+    <InnerLink {...props} forwardedRef={ref} />
+));
+
+// Those two way of writing are equivalent
+export const Link = forwardRef<InnerLinkProps, Element>((props, ref) => (
+    <InnerLink {...props} forwardedRef={ref} />
+));
+```
+
+Example of the way the link could be exported to properly type check the RouterLink's properties.
+
+```jsx
+import { forwardRef } from "@orbit-ui/react-components";
+
+export const MyRouterLink = forwardRef<InnerLinkProps, RouterLink>((props, ref) => (
+    <InnerLink {...props} forwardedRef={ref} />
+));
+```
+
+The `forwardRef` defined in @orbit-ui/react-components returns an [OrbitComponent](orbitcomponent) instead of the standard ForwardRefExoticComponent from React.
+
+## OrbitComponent
+
+The `OrbitComponent` is a type that is used to mark specific components as Orbit components rather than using ForwardRefExoticComponent.
+This is because an `OrbitComponent` comes with the specific props of the component and adds properties extracted from the as property type. It also removes the forwardedRef property declared in the Inner[X]Props interface, which is a standard pattern in Orbit.
+
+`OrbitComponent` takes 2 type generic, the props of the component, and the specific type of the as property. 
+
+The type property could be one of the following : 
+- an element type the element type (like "div", "button", etc), 
+- a React component Type
+- Element, or any of its subtype.
```

Comment:
Honestly I don't think I would understand this whole OrbitComponent section if is was not previously aware of what you did ;)

Reply:
Would also be nice to have an example oh how `OrbitComponent` can be used without `forwardRef`. Maybe a similar exemple for `ForwardRefExoticComponent` ?